### DISCUSSION TITLE
a command to return scanned files

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4927] [Q&A]
let's say using this code

```
import duckdb
import pyarrow.dataset as ds

# Open dataset 
table = ds.dataset('S3://bucket') # Many parquet files
con = duckdb.connect(' select * from table where x=1)
```

is there a command where I can list all the files scanned by DuckDB ? 

my idea is get the the list of files, download it into a local SSD and then run the Query, ideally that command will just show the scanned files, not execuste the whole Query ?

in an ideal World DuckDB can have an option to keep a cached copy of the files.

**Comment by Mytherin** [comment]
The `glob` function should do what you want:

```sql
SELECT * FROM glob('*.parquet');
```

**Reply by djouallah** [reply]
sorry I was not clear, 
I want the list of  files used in an arbitrary Query, not all the files ? if a Query scan only 1 file from a bucket of 1000 files, I want only 1 file.
